src/main/java/mate/academy/spring/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,34 @@
+package mate.academy.spring.security;
+
+import java.util.Optional;
+import mate.academy.spring.model.User;
+import mate.academy.spring.service.UserService;
+import org.springframework.security.core.userdetails.User.UserBuilder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserService userService;
+
+    public CustomUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        Optional<User> optionalUser = userService.findByEmail(email);
+        if (optionalUser.isEmpty()) {
+            throw new UsernameNotFoundException("User not found");
+        }
+        User user = optionalUser.get();
+        UserBuilder userBuilder =
+                org.springframework.security.core.userdetails.User.withUsername(user.getEmail());
+        userBuilder.password(user.getPassword());
+        userBuilder.roles(user.getRoles().stream()
+                .map(r -> r.getRoleName().getName()).toArray(String[]::new));

```

Comment:
does it work? I guess, you needed to add prefix `ROLE_` to every roles' names, so it could work